pkgs/jnigen/lib/src/bindings/dart_generator.dart

```diff
@@ -413,7 +413,10 @@ class $name$typeParamsDef extends $superName {
       }
       s.write('''
     } catch (e) {
-      return $_protectedExtension.newDartException(e.toString());
+      if (e is jni.JObject && jni.Jni.env.IsInstanceOf(e.reference.pointer, jThrowableClass.reference.pointer)) {
```

Comment:
Maybe this logic could reside in `ProtectedJniExtensions` as well so we can just generate:

```dart
catch(e) {
  return $_protectedExtension.newDartException(e);
}
```

And the method does the checks and `toString` and all.

Reply:
That sounds good! And, done ✅